controllers/provider-azure/pkg/controller/controlplane/valuesprovider.go

```diff
@@ -196,6 +197,13 @@ func getConfigChartValues(
 		return nil, errors.Wrapf(err, "could not determine subnet, availability set, route table or security group name from infrastructureStatus of controlplane '%s'", util.ObjectName(cp))
 	}
 
+	var maxNodesCount int32
+	maxNodesCount = 0
```

Comment:
The default value of int32 is already 0. I guess there is no need from this statement.